src/apps/api/views/submissions.py

```diff
@@ -96,6 +99,37 @@ def has_admin_permission(self, user, submission):
         competition = submission.phase.competition
         return user.is_superuser or user in competition.all_organizers
 
+    @action(detail=True, methods=('POST', 'DELETE'))
+    def submission_leaderboard_connection(self, request, pk):
+        submission = self.get_object()
+        phase = submission.phase
+
+        if not (request.user.is_superuser or request.user == submission.owner or phase.competition.collaborators.filter(pk=request.user.pk).exists()):
+            raise Http404
+
+        if request.method == 'POST':
+            # Removing any existing submissions on leaderboard
+            Submission.objects.filter(phase=phase, owner=submission.owner).update(leaderboard=None)
+
+            leaderboard = submission.phase.leaderboard
+
+            if submission.has_children:
+                for s in Submission.objects.filter(parent=pk):
```

Comment:
Minor nitpick here: use `submission` rather than `pk`. Makes it ever-so-slightly more readable.